c2corg_ui/static/js/advancedsearch.js

```diff
@@ -0,0 +1,78 @@
+goog.provide('app.AdvancedSearchController');
+goog.provide('app.advancedSearchDirective');
+
+goog.require('app');
+
+
+/**
+ * This directive is used to display the advanced search form and results.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.advancedSearchDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppAdvancedSearchController',
+    controllerAs: 'searchCtrl',
+    bindToController: true,
+    scope: true,
+    templateUrl: '/static/partials/advancedsearch.html'
+  };
+};
+
+
+app.module.directive('appAdvancedSearch', app.advancedSearchDirective);
+
+
+/**
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {app.Api} appApi Api service.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.AdvancedSearchController = function($attrs, appApi) {
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.doctype_ = $attrs['doctype'];
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {number}
+   * @export
+   */
+  this.total = 0;
+
+  /**
+   * @type {Array.<Object>}
+   * @export
+   */
+  this.documents = [];
+
+  this.getResults_();
+};
+
+
+/**
+ * @private
+ */
+app.AdvancedSearchController.prototype.getResults_ = function() {
+  this.api_.listDocuments(this.doctype_).then(function(response) {
```

Comment:
Please note this is quite outdated code. See https://github.com/c2corg/v6_ui/blob/advanced-search-ui/c2corg_ui/static/js/advancedsearch.js#L108-L116 instead (even though the way it works is close).

> This could be simplified by having listDocuments directly return the data.

I don't understand what you mean. The way ``listDocuments`` is consistent with the other ``apiService`` methods. The success callback is quite specific to the advanced search and cannot be outsourced to the api service.

> On the model of appx.SearchDocumentResponse, the data and this.documents could be typed.

You mean I should create a ``appx.SearchDocumentResponse`` precising the response types as for ``appx.SimpleSearchDocumentResponse`` or maybe reuse it?